### PR TITLE
Cleanup fragile NGen leftovers

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
Follow up from loongarch code review

**Review by BruceForstall — APPROVED** [review]
LGTM

File: src/coreclr/vm/i386/asmhelpers.S [hunk 1]
```diff
@@ -663,47 +663,6 @@ LOCAL_LABEL(GoCallCalliWorker):
...
-    call        C_FUNC(ExternalMethodFixupWorker)
-
-    // eax now contains replacement stub. PreStubWorker will never return
-    // NULL (it throws an exception if stub creation fails.)
-
-    // From here on, mustn't trash eax
-
-    STUB_EPILOG
-
-PATCH_LABEL ExternalMethodFixupPatchLabel
```
Comment (am11) [inline src/coreclr/vm/i386/asmhelpers.S:698]: Should this be deleted this as well:
https://github.com/dotnet/runtime/blob/66eb69ed282c2d35fc64f94e76cf3eeaf83edb5c/src/coreclr/vm/i386/asmhelpers.asm#L1290-L1291

File: src/coreclr/vm/arm/asmhelpers.S [hunk 2]
```diff
@@ -936,8 +899,9 @@ DelayLoad_MethodCall:
 
     EPILOG_WITH_TRANSITION_BLOCK_TAILCALL
 
-    // Share the patch label
-    b C_FUNC(ExternalMethodFixupPatchLabel)
+    EPILOG_WITH_TRANSITION_BLOCK_TAILCALL
```
Comment (jkotas) [inline src/coreclr/vm/arm/asmhelpers.S:902]:
```suggestion
```

**Comment by jkotas** [comment]
We do use PGO data for optimizations, but this specific path is probably not used anymore. @davidwrighton will know for sure.

**Pull request merged**